test/unit/compiler/passes/generate-bytecode.spec.js

```diff
@@ -32,11 +32,27 @@ describe("compiler pass |generateBytecode|", () => {
         "a = 'a'",
         "b = 'b'",
         "c = 'c'",
+        "d = '__proto__'",
+        "e = 'constructor'",
+        "f = 'hasOwnProperty'",
+        "g = 'isPrototypeOf'",
+        "h = 'propertyIsEnumerable'",
+        "i = 'toLocaleString'",
+        "j = 'toString'",
+        "k = 'valueOf'",
```

Comment:
Probably it is better to create a dedicated test with explicit intention to check that special properties are allowed.

Reply:
I'll squash the two last commits if this change works for you @Mingun